test/e2e/persistent_volumes.go

```diff
@@ -41,110 +71,150 @@ func nfsServerPodCleanup(c *client.Client, config VolumeTestConfig) {
 	if config.serverImage != "" {
 		podName := config.prefix + "-server"
 		err := podClient.Delete(podName, nil)
-		if err != nil {
-			framework.Logf("Delete of %v pod failed: %v", podName, err)
-		}
+		Expect(err).NotTo(HaveOccurred())
 	}
 }
 
-// Delete the PV. Fail test if delete fails. If success the returned PV should
-// be nil, which prevents the AfterEach from attempting to delete it.
-func deletePersistentVolume(c *client.Client, pv *api.PersistentVolume) (*api.PersistentVolume, error) {
+// Cleanup up pvs and pvcs in multi-pv-pvc test cases. All entries found in the pv and
+// claims maps are deleted.
+// Note: this is the only code that deletes PV objects.
+func pvPvcCleanup(c *client.Client, ns string, pvols pvmap, claims pvcmap) {
+
+	if c != nil && len(ns) > 0 {
+		framework.Logf("AfterTest: deleting %v PVCs from map", len(claims))
+		for pvcKey := range claims {
+			delete(claims, pvcKey) // ok to delete key from map now
+			Expect(len(pvcKey)).NotTo(BeZero())
+			nSpace := pvcKey.NS()
+			name := pvcKey.Name()
+			_, err := c.PersistentVolumeClaims(nSpace).Get(name)
+			if !apierrs.IsNotFound(err) {
+				Expect(err).NotTo(HaveOccurred())
```

Comment:
If the failure occurs in the normal path of deleting the claim to trigger the recycler (and to test that the related pv becomes Available) the remaining pvcs will be deleted in the AfterEach. However, if the failure occurs in the AfterEach itself, then I guess the remaining claims are not deleted. I am not certain how Expect() works inside an AfterEach. Do you know of a way to harden this code?